astropy/convolution/setup_package.py

```diff
@@ -0,0 +1,24 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import os
+from numpy import get_include as get_numpy_include
+from distutils.extension import Extension
+from astropy_helpers.openmp_helpers import add_openmp_flags_if_available
+
+C_CONVOLVE_PKGDIR = os.path.relpath(os.path.dirname(__file__))
+
+SRC_FILES = [os.path.join(C_CONVOLVE_PKGDIR, filename)
+              for filename in ['boundary_none_direct.c',
+                               'boundary_padded_direct.c']]
+
+def get_extensions():
+    # Add '-Rpass-missed=.*' to ``extra_compile_args`` when compiling with clang
+    # to report missed optimizations
+    c_convolve_ext = Extension(name='c_convolve', sources=SRC_FILES,
+                 extra_compile_args=['-O3', '-fPIC'],
```

Comment:
It is, for vectorization and other such magic. I can turn them all on individually though. Are you concerned about fp arithmetic issues, i.e. fast-math?

Reply:
Actually it looks like conda's Python uses `-ftree-vectorize` and `-O3` by default, pyenv's Python has `-O3` but not `-ftree-vectorize`, Debian's Python is more conservative with `-O2`. So actually my point was more about not specifying additional options here to inherit the default ones instead.

About fast-math, I think it is not activated with `-O3`, I guess it could be interesting here since you are interpolating the NaNs before calling the C code, but not sure if it can lead to other issues.

Reply:
Though this is a very old issue, since I came across this while looking through performance->compiler flags->openmp issues, I will note that it used to be the case that `-O3` was `-O2 -ftree-vectorize` but that seems to be no longer the case. Release notes from gcc14 - https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Optimize-Options.html#index-O3